Makefile

```diff
@@ -0,0 +1,68 @@
+VERSION := 0.16.5
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect sources
+ifneq ($(AMALGAMATED),1)
+	SRC := $(wildcard lib/src/*.c)
+	# do not double-include amalgamation
+	SRC := $(filter-out lib/src/lib.c,$(SRC))
+else
+	# use amalgamated build
+	SRC := lib/src/lib.c
+endif
+OBJ := $(SRC:.c=.o)
+
+# define default flags, and override to append mandatory flags
+CFLAGS ?= -O3
+override CFLAGS += -std=gnu99 -fPIC -Ilib/src -Ilib/include
+LDFLAGS ?=
+override LDFLAGS +=
+
+# ABI versioning
+SONAME_MAJOR := 1
```

Comment:
Minor nit: Can we start at SOVERSION 0?  That seems to be more standard, and will hopefully make it more obvious that the shared library version string has ~nothing to do with the project semver version number.

Reply:
(Unless the SOVERSION 1 is intended to bump past any SOVERSION-0 packages that might have already been built?)

Reply:
I don't know how standard this is, so I'll go with whatever @maxbrunsfeld wants.

Reply:
I would say let’s start at 0 like @dcreager said.

Reply:
Okay, libtree-sitter.so.0.0 it is.

Reply:
Done.